src/imitation/algorithms/adversarial.py

```diff
@@ -133,12 +134,17 @@ def __init__(self,
           self.venv, self.reward_test)
 
     self.venv_train_norm = VecNormalize(self.venv_train)
+    self.venv_train_norm_buffering = BufferingWrapper(self.venv_train_norm)
 
     if gen_replay_buffer_capacity is None:
       gen_replay_buffer_capacity = 20 * self.gen_batch_size
     self._gen_replay_buffer = buffer.ReplayBuffer(gen_replay_buffer_capacity,
                                                   self.venv)
-    self._populate_gen_replay_buffer()
+    gen_samples = rollout.generate_transitions(
+      self.gen_policy,
+      self.venv_train_norm,
+      n_timesteps=self.gen_batch_size // 2)
+    self._gen_replay_buffer.store(gen_samples)
```

Comment:
Is this just to make sure the replay buffer isn't empty and so can be sampled immediately? I wonder if we can avoid this by doing `train_gen` before `train_disc`?

Reply:
Yes, that would work. But not prepopulating the replay buffer has another problem where we have to either
(1) error if the user calls`train_disc` is called before `train_gen` or 
(2) prepopulate the replay buffer if `train_disc` is called before `train_gen`

In my head (2) seems like the preferable solution to me in which case we might as well simplify the code by prepopulating upon `__init__`.

Reply:
We decided to error if `train_disc` is called before `train_gen` is called.